apps/movies/src/movies.controller.ts

```diff
@@ -28,8 +31,25 @@ export class MoviesController {
   }
 
   @Get()
-  findAll() {
-    return this.moviesService.findAll();
+  async getPaginatedMovies(
+    @Query('page', new DefaultValuePipe(1), ParseIntPipe) page = 1,
+    @Query('limit', new DefaultValuePipe(20), ParseIntPipe) limit = 20
+  ): Promise<PaginationResponseDto<MovieResponseDto[]>> {
+    limit = limit > 100 ? 100 : limit;
+
+    const result = await this.moviesService.findAll({
+      page,
+      limit,
+      route: 'http://localhost:3001/movies',
+    });
```

Comment:
```suggestion
    }, searchOptions);
```

You can check how does searchOptions look like if you click on paginate method from lib